src/services/import/zip.ts

```diff
@@ -88,30 +86,30 @@ async function importZip(taskContext, fileBuffer, importRootNote) {
         return {};
     }
 
-    /** @returns {{noteMeta: NoteMeta|undefined, parentNoteMeta: NoteMeta|undefined, attachmentMeta: AttachmentMeta|undefined}} */
-    function getMeta(filePath) {
+    function getMeta(filePath: string) {
         if (!metaFile) {
             return {};
         }
 
         const pathSegments = filePath.split(/[\/\\]/g);
 
-        /** @type {NoteMeta} */
-        let cursor = {
+        let cursor: NoteMeta | undefined = {
             isImportRoot: true,
-            children: metaFile.files
+            children: metaFile.files,
+            dataFileName: ""
         };
 
-        /** @type {NoteMeta} */
-        let parent;
+        let parent!: NoteMeta;
```

Comment:
Same here, I'm curious what the benefit is over:
```suggestion
        let parent: NoteMeta | undefined;
```

Reply:
@j-tirelli , I don't think there's any benefit to using the `!` operator. I used it to avoid the errors about using a value before it's declared. I have to admit the code in `zip.ts` was a bit hard to follow for me, so I just did what was necessary to avoid any build errors. Nevertheless, the `| null = null` approach is much cleaner since it provides better type checking, even if it requires adding some new null checks here and there.